internal/controller/model_nodeport.go

```diff
@@ -40,27 +52,61 @@ func NewNodePortLoadBalancerModelGenerator(
 	}
 }
 
-func (g *NodePortLoadBalancerModelGenerator) getDestinationAddresses() (result []string, err error) {
+func isValidIp(ipString string) (err error) {
+	ipParsed := net.ParseIP(ipString)
+	if ipParsed != nil {
+		return nil
+	}
+	return ErrInvalidIpAddress
+}
+
+func isIPv4Address(ipString string) (isIPv4 bool, err error) {
+	return strings.Count(ipString, ":") < 2, isValidIp(ipString)
+}
+
+func isIPv6Address(ipString string) (isIPv6 bool, err error) {
+	return strings.Count(ipString, ":") >= 2, isValidIp(ipString)
+}
```

Comment:
I think the `is*Address` functions do a bit too much. This is apparent from the error handling required below.

I think it would be cleaner to simply assume that a string is an IP address and require the caller to check for "being an IP address" first.

Reply:
I agree, I simplified the functions and error handling.